book/src/protocol.md

```diff
@@ -0,0 +1,611 @@
+
+$$
+\sum_{j=0}^{k - 1} u_j^{-1} [L_j]^{\mathbf{G}}_i + [P']^{\mathbf{G}}_i + \sum_{j=0}^{k - 1} u_j [R_j]^{\mathbf{G}}_i = c \mathbf{s}_i \forall i \in [0, n)
+$$
+
+and the equality
+
+$$
+\sum_{j=0}^{k - 1} u_j^{-1} [L_j]^U + [P']^U + \sum_{j=0}^{k - 1} u_j [R_j]^U = cz \prod_{j=0}^{k - 1} (1 + u_{k - 1 - j} x_3^{2^j}).
+$$
+
+Recall that all of these equalities hold or else $\mathcal{F}$ will return a non-trivial discrete log relation. Substituting for $c$, we obtain the following $n$ equalities
+
+$$
+\begin{array}{ll}
+&\sum\limits_{j=0}^{k - 1} u_j^{-1} [L_j]^U + [P']^U + \sum\limits_{j=0}^{k - 1} u_j [R_j]^U \\
+= &s_i^{-1} z \prod\limits_{j=0}^{k - 1} (1 + u_{k - 1 - j} x_3^{2^j}) \\
+&\cdot \left( \sum\limits_{j=0}^{k - 1} u_j^{-1} [L_j]^{\mathbf{G}}_i + [P']^{\mathbf{G}}_i + \sum\limits_{j=0}^{k - 1} u_j [R_j]^{\mathbf{G}}_i \right)\\
+&\forall i \in [0, n)
+\end{array}
+$$
+
+If we let $r = k - 1$ we can rewrite the above as
+
+$$
+\begin{array}{ll}
+&\sum\limits_{j=0}^{r} u_j^{-1} [L_j]^U + [P']^U + \sum\limits_{j=0}^{r} u_j [R_j]^U \\
+= &s_{2^{k - 1 - r} i}^{-1} z \prod\limits_{j=k - 1 - r}^{k - 1} (1 + u_{k - 1 - j} x_3^{2^j}) \\
+&\cdot \left(
+  \sum\limits_{j=0}^{r} \left(
+    \sum\limits_{h=0}^{2^{k - 1 - r} - 1} x_3^h \left(
+      u_j^{-1} [L_j]^{\mathbf{G}}_{2^{k - 1 - r}i + h} + u_j [R_j]^{\mathbf{G}}_{2^{k - 1 - r}i + h}
+    \right)
+  \right)
++ \sum\limits_{h=0}^{2^{k - 1 - r} - 1} x_3^h \left(
+    [P']^{\mathbf{G}}_{2^{k - 1 - r}i + h}
+  \right)
+\right) \\
+&\forall i \in [0, 2^{r + 1})
+\end{array}
+$$
+
+By substituting and simplifying it can be seen that the former expression is a special case of the latter (where $r = k - 1$). We will now show that when these expressions hold for the case of $r \geq 0$, they hold for the case of $r - 1$ also.
+
+It is the case that for $k - 1 \geq r \geq 0$ we have that $\mathbf{s}_{2^{k - r}i + 2^{k - 1 - r}} = \mathbf{s}_{2^{k - r}i} \cdot u_r$ for all $i \in [0, 2^r)$. This allows us to relate the odd and even values of $i$ in the above equalities.
+
+snip
+
+Observe that in the base case of $r = -1$ where one equality holds ($i = 0$), we obtain
+
+$$
+\begin{array}{ll}
+[P']^U &= s_{2^k i}^{-1} z \cdot \sum\limits_{h=0}^{2^k - 1} x_3^h \left(
+    [P']^{\mathbf{G}}_{2^k i + h}
+  \right) \\
+&= z \cdot \sum\limits_{h=0}^{n-1} x_3^h [P']^{\mathbf{G}}_h
+\end{array}
+$$
+
+Because $[P']$ was determined prior to the choice of $z$, for nonzero $[P']^U$ this equality would hold with probability $\frac{1}{|\ch|}$ and so by bounding $|\badch_{\tr'_z}| \leq \frac{1}{|\ch|} \leq \epsilon$ we can conclude $[P']^U = \sum\limits_{h=0}^{n-1} x_3^h [P']^{\mathbf{G}}_h = 0$ and therefore that $[P']^{\mathbf{G}}$ describes a $n - 1$ degree polynomial that has a root at $x_3$.
```

Comment:
I think this is slightly better in that it doesn't break across a line:
```suggestion
Because $[P']$ was determined prior to the choice of $z$, for nonzero $[P']^U$ this equality would hold with probability $\frac{1}{|\ch|}$ and so by bounding $|\badch_{\tr'_z}| \leq \frac{1}{|\ch|} \leq \epsilon$ we can conclude $$[P']^U = \sum\limits_{h=0}^{n-1} x_3^h [P']^{\mathbf{G}}_h = 0$$ and therefore that $[P']^{\mathbf{G}}$ describes a $n - 1$ degree polynomial that has a root at $x_3$.
```